pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go

```diff
@@ -105,6 +65,9 @@ type CollectorRunConfig struct {
 	Config                      *config.OrchestratorConfig
 	MsgGroupRef                 *atomic.Int32
 	OrchestratorInformerFactory *OrchestratorInformerFactory
+
+	// used by ecs collector
+	WorkloadmetaStore workloadmeta.Component
```

Comment:
I think it'd be nice to create and embed a K8sCollectorRunConfig and ECSCollectorRunConfig struct to have separation of common config and orchestrator-specific configs.